x-pack/filebeat/module/fortinet/firewall/ingest/pipeline.yml

```
@@ -47,12 +47,12 @@ processors:
     ignore_empty_value: true
 - set:
```

Should `fortinet.firewall.tz` overwrite `event.timezone` if present?
```suggestion
- set:
    field: event.timezone
    value: "{{fortinet.firewall.tz}}"
    if: "ctx.fortinet?.firewall?.tz == null"
- set:
```

And then use only `event.timezone` in all the `date` processors in this pipeline.

This way, if the log has a timezone, this one is used, if it hasn't, https://github.com/elastic/beats/issues/20300#issuecomment-734404378 applies.

---

This is kinda what I was thinking. However I assume u mean `ctx.fortinet?.firewall?.tz != null`

---

Yes! Sorry, good catch :slightly_smiling_face: I have updated my comment, thanks!

---

This was already in the pipeline in https://github.com/elastic/beats/pull/26524/files#diff-a7f0621e4d2d62110063c53a6b6b9afa686129f641a47e541a44d55bd648fb54L44 👍